### ISSUE TITLE
set min_child_weight as a float instead of int

### ISSUE THREAD

**Issue by QuantHao** [dmlc/xgboost#5976]
Hi, all

Based on xgboost documentation, min_child_weight is treated as number of observation for rmse objective and is usually chosen as int such as https://xgboost.readthedocs.io/en/latest/python/python_api.html#xgboost.XGBRegressor.

However, I recently read a article saying that "With XGBoost, you can specify this parameter as a float as well. If you do, it will use a percentage of samples to determine leaf nods." (https://kevinvecmanis.io/machine%20learning/hyperparameter%20tuning/dataviz/python/2019/05/11/XGBoost-Tuning-Visual-Guide.html#child_weight). I cannot find a second source using min_child_weight as a float.

So I am confused and worried that is it true that we could set min_child_weight as a float?? Any comfirmation would be appreciated.

**Comment by QuantHao** [comment]
sklearn's min_samples_leaf supports float as a percentage of total sample size which makes sense to me. It would be great if xgboost supports similar functionality.

**Comment by trivialfis** [comment]
Sorry for the mistake in our doc.   Also, the parameter you referred `min_samples_leaf` is indeed on our roadmap: https://github.com/dmlc/xgboost/issues/5444

**Comment by QuantHao** [comment]
Hi @trivialfis , thanks so much for your timely reply. May I double check with you that if min_child_weight is set as, for example, 0.01, it will be treated as n_sample * 0.01, meaning 1% of total sample?

Reference: https://kevinvecmanis.io/machine%20learning/hyperparameter%20tuning/dataviz/python/2019/05/11/XGBoost-Tuning-Visual-Guide.html#child_weight

**Comment by trivialfis** [comment]
No.  It's the minimum value of accumulated Hessian for each leaf.  Hessian is often used as a proxy for data in gradient boosting.

**Comment by QuantHao** [comment]
> No. It's the minimum value of accumulated Hessian for each leaf. Hessian is often used as a proxy for data in gradient boosting.

I see. Thanks for answering.
